Chat/AbstractSocket.cpp

```diff
@@ -0,0 +1,30 @@
+#include "AbstractSocket.h"
+
+AbstractSocket::AbstractSocket() : socket_(INVALID_SOCKET)
+{}
+
+AbstractSocket::~AbstractSocket(){}
+
+int AbstractSocket::Initialize(int type)
```

Comment:
I would rather give a name to your enum (sth like enum ConnectionType{ TCP, UDP}). If so, Initialize should take ConnectionType parameter, not int. But it will be better if you set "const ConnectionType& type" there.
Also I would rather make that function boolean(false if INVALID_SOCKET and true if not).